packages/core/src/apiml/Services.ts

```diff
@@ -0,0 +1,244 @@
+*
+*/
+
+import { AbstractSession, IConfig, IConfigProfile, ImperativeConfig, ImperativeExpect, Logger,
+         PluginManagementFacility, RestClient } from "@zowe/imperative";
+import { ApimlConstants } from "./ApimlConstants";
+import { IApimlProfileInfo } from "./doc/IApimlProfileInfo";
+import { IApimlService } from "./doc/IApimlService";
+import { IApimlSvcAttrsLoaded } from "./doc/IApimlSvcAttrsLoaded";
+import * as JSONC from "comment-json";
+import { ConfigConstants } from "../../../../../imperative/lib";
+
+/**
+ * Class to handle listing services on APIML gateway.
+ * @export
+ * @class Services
+ */
+export class Services {
+    /**
+     * Forms a list of APIML service attributes needed to query APIML for every
+     * REST service for every loaded command group. This information can later
+     * be used to create every connection profile required for every loaded
+     * command group.
+     *
+     * @returns The APIML service attributes needed to query APIML.
+     */
+    public static getPluginApimlConfigs(): IApimlSvcAttrsLoaded[] {
+        const apimlConfigs: IApimlSvcAttrsLoaded[] = [];
+        ImperativeExpect.toNotBeNullOrUndefined(ImperativeConfig.instance.loadedConfig,
+            "Imperative.init() must be called before getPluginApimlConfigs()"
+        );
+
+        // Get the APIML configs from the loaded imperative config
+        for (const apimlConfig of (ImperativeConfig.instance.loadedConfig.apimlConnLookup || [])) {
+            apimlConfigs.push({
+                ...apimlConfig,
+                connProfType: apimlConfig.connProfType || ImperativeConfig.instance.loadedConfig.profiles[0].type,
+                pluginName: ImperativeConfig.instance.hostPackageName
+            });
+        }
+
+        // Load APIML configs from all plugins
+        for (const pluginCfgProps of PluginManagementFacility.instance.allPluginCfgProps) {
+            for (const apimlConfig of (pluginCfgProps.impConfig.apimlConnLookup || [])) {
+                apimlConfigs.push({
+                    ...apimlConfig,
+                    connProfType: apimlConfig.connProfType || pluginCfgProps.impConfig.profiles[0].type,
+                    pluginName: pluginCfgProps.pluginName
+                });
+            }
+        }
+        return apimlConfigs;
+    }
+
+    /**
+     * Calls the services endpoint of the APIML gateway to obtain a list of
+     * services that support Single Sign-On. This list is compared against a
+     * list of APIML service attributes defined in CLI plug-in configs. When a
+     * service's API ID is present in both lists, a profile info object is
+     * generated to store CLI profile info for connecting to that service.
+     * @static
+     * @param session Session with APIML connection info
+     * @param configs APIML service attributes defined by CLI plug-ins
+     * @returns List of objects containing profile info for APIML services
+     * @memberof Services
```

Comment:
Should we document the errors that this function can throw (e.g. `@throw [XYZ error] when Session is undefined`)?
The `[XYZ error]` portion may not be necessary since they will all be `ImperativeError`s